core/arch/arm/mm/core_mmu.c

```diff
@@ -89,7 +89,7 @@ register_phys_mem_ul(MEM_AREA_TEE_RAM_RX, VCORE_UNPG_RX_PA, VCORE_UNPG_RX_SZ);
 register_phys_mem_ul(MEM_AREA_TEE_RAM_RO, VCORE_UNPG_RO_PA, VCORE_UNPG_RO_SZ);
 
 #ifdef CFG_VIRTUALIZATION
-register_phys_mem_ul(MEM_AREA_TEE_RAM_RO, VCORE_UNPG_RW_PA, VCORE_UNPG_RW_SZ);
+register_phys_mem_ul(MEM_AREA_TEE_RAM_RW, VCORE_UNPG_RW_PA, VCORE_UNPG_RW_SZ);
```

Comment:
This change is not  needed. What we are doing there is mapping of "golden" .data section. This memory will be then copied to .data section for new guest instance. So we don't want to make it read-write.

Reply:
@lorc and all, would you mind if we add an inline comment like `/* Map unpaged RW memory as RO, will be copied to .data for new guest instances */` ?

Reply:
Yes, that would be great. Actually, I wanted to suggest this.